### DISCUSSION TITLE
为什么backend模型初始化的时候，CLASSES和PALETTE要从data type的配置获取，而不是从checkpoint加载。

### DISCUSSION THREAD

**Discussion by liuyunfeng2016** [open-mmlab/mmdeploy#198] [Q&A] [answered]
https://github.com/open-mmlab/mmdeploy/blob/34879e638cc2db511e798a376b9a4b9932660fe1/mmdeploy/codebase/mmseg/deploy/segmentation_model.py#L195
因为一些自定义的dataset type并没有确定这里的值，只在初始化之后，才获取到这两个类的值。这个是不是应该和初始化pytorch model保持一致，直接从checkpoint中读取更好

**Comment by grimoire** [comment, marked as answer]
我向其他repo的开发者确认了一下，事实上现在的实现里并没有办法保证CLASSES或PALLETTE等属性一定存在于checkpoint中。

**Reply by liuyunfeng2016** [reply]
好吧，那就需要自定义的类，像官方一样，将全局变量的值的类别给定死，感谢你的回复
